pynamodb/tests/test_attributes.py

```diff
@@ -707,17 +722,69 @@ def __eq__(self, other):
         person1 = Person()
         person1.name = 'john'
         person1.age = 40
+        person1.extra_arbitrary_data = {'married': True}
 
         person2 = Person()
         person2.name = 'Dana'
         person2.age = 41
+        person2.extra_arbitrary_data = {'net_income': 200000,
+                                        'dog': 'Fido'}
+
         inp = [person1, person2]
 
         list_attribute = ListAttribute(default=[], of=Person)
         serialized = list_attribute.serialize(inp)
         deserialized = list_attribute.deserialize(serialized)
         self.assertEqual(sorted(deserialized), sorted(inp))
 
+    def test_list_of_map_with_of_and_custom_attribute(self):
+
+        # Create a couple of mock functions to use
+        # to test that the CustomAttribute serialize/deserialize are called
+        serialize_mock = Mock()
```

Comment:
An alternative way of doing this is by using `attribute1 = Mock(wraps=CustomMapAttribute())`. This will give you something that acts like a `CustomMapAttribute` but you can also inspect method calls in the usual mock way (i.e `attribute1.serialize.called`, etc.)